modules/openthread/platform/crypto_psa_key_allocator.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef OPENTHREAD_PLATFORM_CRYPTO_PSA_KEY_ALLOCATOR_H_
+#define OPENTHREAD_PLATFORM_CRYPTO_PSA_KEY_ALLOCATOR_H_
+
+#include <psa/crypto.h>
+#include <openthread/platform/crypto.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * A macro helper for using the getKeyRef function.
+ * This macro automatically returns the error code if the function fails and resets the attributes
+ * if provided.
+ */
+#define GET_KEY_REF(keyRef, attributes)                                                            \
+	do {                                                                                       \
+		otError err = getKeyRef((keyRef), (attributes));                                   \
+		if (err != OT_ERROR_NONE) {                                                        \
+			if (attributes) {                                                          \
+				psa_reset_key_attributes(attributes);                              \
+			}                                                                          \
+			return err;                                                                \
```

Comment:
(also noticed that the `err` is not checked anywhere)

Reply:
But it is a macro, so it returns an error directly to the function where it is used, so it is checked in the upper layer.